pkg/cache/internal/informers.go

```diff
@@ -290,7 +310,12 @@ func (ip *Informers) Get(ctx context.Context, gvk schema.GroupVersionKind, obj r
 		}
 	}
 
-	if started && !i.Informer.HasSynced() {
+	cfg := &GetOptions{}
+	for _, opt := range opts {
+		opt(cfg)
+	}
+
+	if started && !i.Informer.HasSynced() && !cfg.DoNotBlockUntilSynced {
```

Comment:
Can we add a test for this, please?

Reply:
Yeah, to avoid duplicate work, I'm waiting for agreement on form before dedicating time to tests.

Reply:
Added unit tests